cli/js/testing.ts

```diff
@@ -62,70 +62,146 @@ export function test(
   TEST_REGISTRY.push(testDef);
 }
 
-interface TestStats {
-  filtered: number;
-  ignored: number;
-  measured: number;
-  passed: number;
-  failed: number;
-}
-
-export interface RunTestsOptions {
-  exitOnFail?: boolean;
-  failFast?: boolean;
-  only?: string | RegExp;
-  skip?: string | RegExp;
-  disableLog?: boolean;
-  reporter?: TestReporter;
-}
-
-enum TestStatus {
-  Passed = "passed",
-  Failed = "failed",
-  Skipped = "skipped"
-}
-
-interface TestResult {
-  name: string;
-  status: TestStatus;
-  duration: number;
-  error?: Error;
-}
-
-export enum TestEvent {
+enum TestEvent {
   Start = "start",
   TestStart = "testStart",
   TestEnd = "testEnd",
   End = "end"
 }
 
-interface TestEventStart {
-  kind: TestEvent.Start;
-  tests: number;
-}
+// Namespace containing lesser used testing APIs.
+// test(), TestDefinition, runTests() and RunTestsOptions should stay top-level.
+// eslint-disable-next-line @typescript-eslint/no-namespace
+export namespace tests {
```

Comment:
The error that comes up when I don't disable this lint rule says to use ES modules instead of namespaces. Another problem is that `tests` when declared like this is treated differently than the other namespaces in `Deno`, and is even highlighted a different colour by VS Code. I think I'll create a `cli/js/tests.ns.ts`.

Reply:
But that makes a dependency cycle 😭